src/microsim/devices/MSDevice_GLOSA.cpp

```diff
@@ -118,37 +124,149 @@ MSDevice_GLOSA::notifyMove(SUMOTrafficObject& /*tObject*/, double oldPos,
+                        }
+#endif
+                        // if increased speed is fast enough to arrive at tls while green (else look for next phases)
+                        if (timetoJunction2 <= (nextSwitch + yellowSlack)) {
+                            // increase speed factor up to a maximum if necessary and useful
+                            // XXX could compute optimal speed factor here
+                            myVeh.setChosenSpeedFactor(myMaxSpeedFactor);
+                            mySpeedAdviceActive = true;
+                            break; // solved
+                        } else {
+                            // speed can not be increased to arrive at tls while green
+                            // next switch is calculated at the end of the for-loop
+                        }
+                    } else {
+                        // speed increase is not enough to reach tls while green
+                        // next switch is calculated at the end of the for-loop
                     }
+                } else {
+                    // vehicle will arrive at tls while green
+                    break; // solved
+                }
+            } else if (currentPhaseStop) {
+                // tls is red at the moment
+#ifdef DEBUG_GLOSA
+                if (DEBUG_COND) {
+                    std::cout << SIMTIME << " veh=" << myVeh.getID() << " tls will be/is red"
+                              << " nextSwitch=" << nextSwitch
+                              << " timeToJunction=" << timeToJunction
+                              << " myDistance=" << myDistance << "\n";
+                }
 #endif
-                    if (timetoJunction2 <= (timeToSwitch + yellowSlack)) {
-                        // increase speed factor up to a maximum if necessary and useful
-                        // XXX could compute optimal speed factor here
-                        myVeh.setChosenSpeedFactor(myMaxSpeedFactor);
-                        mySpeedAdviceActive = true;
+                if (countwhile == 2 && myUseQueue) {
+                    // take queue into account if current green phase can not be passed and next red phase is used instead
+                    nextSwitch += QueueLength * factor + addition;
+#ifdef DEBUG_QUEUE
+                    if (DEBUG_COND) {
+                        std::cout << SIMTIME << " veh=" << myVeh.getID() << " nextSwitch(redadditon): " << nextSwitch << "\n";
                     }
+#endif
+                }
+                adaptSpeed(myDistance, timeToJunction, nextSwitch + myAddSwitchTime, solved);
+                if (countwhile == 2 && myUseQueue) {
+                    // reset queue because following phases should not calculate with current queue
+                    nextSwitch -= QueueLength * factor + addition;
+                    timeToJunction -= additionaljunctiontime;
+                }
+                if (solved == true) {
```

Comment:
no need for the equal operator as the variable is already a boolean

Reply:
Should I change it or will you?

Reply:
You can do it